main.go

```diff
@@ -791,7 +797,9 @@ func stripTenantPrefix(prefix string, next http.Handler) http.Handler {
 			http.Error(w, "tenant not found", http.StatusInternalServerError)
 			return
 		}
-		http.StripPrefix(path.Join("/", prefix, tenant), next).ServeHTTP(w, r)
+
+		tenantPrefix := path.Join("/", prefix, tenant)
+		http.StripPrefix(tenantPrefix, next).ServeHTTP(w, r.Clone(context.WithValue(r.Context(), proxy.TenantPrefixKey, tenantPrefix)))
```

Comment:
Rather than cloning the whole request which is a deep copy AFAIK, why don't we build a new context and use https://golang.org/pkg/net/http/#Request.WithContext to propogate the header?

Reply:
> To create a new request with a context, use NewRequestWithContext. To change the context of a request, such as an incoming request you want to modify before sending back out, use Request.Clone. Between those two uses, it's rare to need WithContext.

I saw this in godoc and decided to use Clone instead. I am not sure if deep copy is necessary here though?

Reply:
Exactly. That's why I'm suggesting to use `Request.WithContext`. `Request.Clone` does the deep clone.

Reply:
https://golang.org/pkg/net/http/#Request.Clone
> Clone returns a deep copy of r with its context changed to ctx. The provided ctx must be non-nil.